### DISCUSSION TITLE
I have a question about console.log

### DISCUSSION THREAD

**Discussion by Merulez** [Cyfrin/foundry-full-course-cu#1149] [Q&A]
Why is it that console.log shows the output only if the test is working properly but if it's failing , it doesn't show the output ???

**Comment by ghost** [comment]
Code runs synchronously, if your `console.log` statement is on line 10 but the function reverts on line 7, the execution stops there and the `console.log` statement never runs, that's why you don't see anything. But on the other way around (`console.log` on line 7 and the function reverts on line 10), you will get both outputs on the terminal: (a) from the `console.log` and (b) some revert reason.
